### PR TITLE
Azure Cloud integration

### PR DESCRIPTION
**Pull request by praveenkumar0566**
After testing internally we observed the file should be cloud-integration.json but not azure-storage-config.json. Also the format of the content in the json file has been updated to the working version

### CODE REVIEW

**Review by mmurph3 — COMMENTED** [review]
LGTM. We also tested this and confirm it works.

**Comment by srpomeroy** [comment]
cloud-integration.json is the new way of doing it as captured in [the following docs.](https://github.com/kubecost/docs/blob/main/multi-cloud.md)

afaik azure-storage-config.json is still a valid way to configure the azure billing integration.
[Here is an existing reference in OpenCost.](https://github.com/opencost/opencost/blob/0094cba41fc0827e1598ceba9f6901739aed6386/pkg/cloud/provider.go#L31)
Used by [this](https://github.com/opencost/opencost/blob/57bda4821a8ac2f91435fd6216d4073006170b89/pkg/cloud/azureprovider.go#L624) function.

Unless something has changed?
cc: @Sean-Holcomb

File: azure-out-of-cluster.md [hunk 1]
```diff
@@ -29,23 +29,27 @@ The values needed to provide access to the Azure Storage Account where cost data
...
+    ]
 }
 ```
 
 Once you have the values filled out use this command to create the secret:
 
-`kubectl create secret generic <SECRET_NAME> --from-file=azure-storage-config.json -n kubecost`
+`kubectl create secret generic <SECRET_NAME> --from-file=cloud-integration.json -n kubecost`
 
 Once the secret is created, set `.Values.kubecostProductConfigs.azureStorageSecretName` to
```
Comment (jessegoodier) [inline azure-out-of-cluster.md:54]: I think we can use 
Values.kubecostProductConfigs.cloudIntegrationSecret: "cloud-integration"
for all providers.

**Comment by jessegoodier** [comment]
One quick suggestion, otherwise I think your changes look good.

**Comment by praveenkumar0566** [comment]
Thanks everyone, can we have this PR merged if it looks good? Looks like i don't have merge permission.

File: azure-out-of-cluster.md (hunk 1 above)
Comment (praveenkumar0566) [reply azure-out-of-cluster.md]: Changed to cloudIntegrationSecre

**Comment by srpomeroy** [comment]
> Thanks everyone, can we have this PR merged if it looks good? Looks like i don't have merge permission.

Per my other comment, I would like the engineering team to comment on whether the azure-storage-config.json is still a valid way to configure before merging.

**Comment by dwbrown2** [comment]
cc @AjayTripathy @Sean-Holcomb

**Comment by jessegoodier** [comment]
> > Thanks everyone, can we have this PR merged if it looks good? Looks like i don't have merge permission.
> 
> Per my other comment, I would like the engineering team to comment on whether the azure-storage-config.json is still a valid way to configure before merging.

Should we just use cloudIntegration unless there's a reason to use the azure specific key?

**Comment by srpomeroy** [comment]
> Should we just use cloudIntegration unless there's a reason to use the azure specific key?
IMO we should be pushing people to use cloud integration for sure. But that doesn't change the fact that this other method may still be supported.

**Comment by thomasvn** [comment]
For your awareness @srpomeroy @jessegoodier 

Although the old method of `.Values.kubecostProductConfig.azureStorageSecretName` may still be supported, it's been shown to no longer function properly in this in this [Github Issue](https://github.com/kubecost/cost-analyzer-helm-chart/issues/1430) and this [internal doc](https://docs.google.com/document/d/1WaUJsmgsyqwSP82RlhETIKmNyaBAvP2PgbVzFJU8BXQ)

**Comment by praveenkumar0566** [comment]
Can someone approve the PR, I need to send the doc to the user with the latest changes.

**Comment by srpomeroy** [comment]
@praveenkumar0566 If your customer needs something immediately, [you can send them the following doc](https://guide.kubecost.com/hc/en-us/articles/4407595968919#azure-1-1) which already covers the Azure cloud integration config. Alternatively, [send them the rendered commit](https://github.com/kubecost/docs/blob/cff22b4be826e4c329cb95fd44a4a72809d847fc/azure-out-of-cluster.md).

I sent a DM to @Sean-Holcomb asking if he could review this PR and conversation.

**Pull request merged**